engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/BpmnStreamProcessor.java

```diff
@@ -88,6 +96,40 @@ public void processRecord(final TypedRecord<ProcessInstanceRecord> record) {
                     record, RejectionType.INVALID_STATE, violation.getMessage()));
   }
 
+  @Override
+  public ProcessingError tryHandleError(
+      final TypedRecord<ProcessInstanceRecord> command, final Throwable error) {
+    if (error instanceof ExceededBatchRecordSizeException) {
+      context.init(
+          command.getKey(), command.getValue(), (ProcessInstanceIntent) command.getIntent());
+      if (context.getBpmnElementType() != BpmnElementType.PROCESS) {
+        // set element's state to what it was doing, this allows us to resolve the incident later
+        final BpmnElementContext transitionedContext;
+        transitionedContext =
+            switch ((ProcessInstanceIntent) command.getIntent()) {
+              case ACTIVATE_ELEMENT -> stateTransitionBehavior.transitionToActivating(context);
+              case COMPLETE_ELEMENT -> stateTransitionBehavior.transitionToCompleting(context);
+              case TERMINATE_ELEMENT -> stateTransitionBehavior.transitionToTerminating(context);
+                // even though we don't know how to resolve this incident, we can still
+                // raise it so the user can use modification to recover. The incident resolution
+                // logic is smart enough to deal with this case. It will log an error due to
+                // IllegalStateException.
+              default -> context;
+            };
+        incidentBehavior.createIncident(
+            new Failure(
+                """
+                Expected to transition element '%s', but ran into MAX_MESSAGE_SIZE limitation. \
```

Comment:
💭 I wonder if it's clearer for users if we say "Expected to process element". Transitioning may be a vague concept for them.

Reply:
💭 If it's easy we could access the MAX_MESSAGE_SIZE here to be more specific about the limitation. But this could also be an improvement at a later stage.
It could also be confusing as the metadata makes it so they can't reach the exact max message size, so might cause confusing than it clarifies.

Reply:
About adding more details about the message size:
- we had this issue as well working on expiring messages in a single command ([comment](https://github.com/camunda/zeebe/pull/13400#discussion_r1269286574))
- I also think these details can be confusing to most users that will read this error message in Operate
- Users have little control over the messages, it's not actionable information
- I prefer to avoid these details